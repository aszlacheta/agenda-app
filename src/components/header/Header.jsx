import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.scss';

export default function Header() {

    const setClassName = ({isActive}) => (isActive ? 'active' : 'inactive')

    return <nav className="header">
        <NavLink to="/agenda" className={setClassName}>Agenda</NavLink>
        <NavLink to="/contact" className={setClassName}>Contact</NavLink>
    </nav>
}