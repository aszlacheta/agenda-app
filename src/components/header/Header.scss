@use '../variables.scss';

.header {
    display: flex;
    height: variables.$headerHeight;
    padding: 0 50px;

    background-color: variables.$headerBackgroundColor;

    a {
        position: relative;
        display: inline-block;
        margin: variables.$headerLinkPadding;

        line-height: variables.$headerHeight;
        text-transform: uppercase;
        letter-spacing: variables.$headerLinkLetterSpacing;
        text-shadow: variables.$headerLinkTextShadow;
        color: variables.$headerLinkColor;
        font-size: variables.$headerLinkFontSize;

        &.active {
            font-family: variables.$headerLinkActiveFontFamily;
        }

        &:after {
            content: '';
            position: absolute;
            bottom: calc(variables.$headerHeight / 4);
            left: calc(variables.$headerLinkLetterSpacing / 2 * -1);
            width: 0%;
            height: variables.$headerLinkHoverLineHeight;
            background-color: variables.$headerLinkColor;
            transition: variables.$headerLinkHoverTransition;
        }
    
        &:hover:after {
            width: 100%;
        }
    }
}
