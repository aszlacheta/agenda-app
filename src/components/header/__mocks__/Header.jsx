import React from 'react';

const Header = () => <div className="mocked-header" />;

module.exports = Header;
