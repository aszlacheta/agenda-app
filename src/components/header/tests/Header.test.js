import 'jsdom-global/register';
import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => {
  return {
    __esModule: true,
    useParams: jest.fn(),
    useHistory: jest.fn(),
    BrowserRouter: ({ children }) => <div className="mocked-browser-router">{children}</div>,
    NavLink: jest.fn().mockReturnValue(<div className="mocked-nav-link" />)
  };
});

describe('Header', () => {
  it('check if renders properly', () => {
    const tree = renderer
      .create(<Header />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('check if contains exactly two links', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.find('.mocked-nav-link').exists()).toBe(true);
  });
});
