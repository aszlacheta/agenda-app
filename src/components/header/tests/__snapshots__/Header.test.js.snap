// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header check if renders properly 1`] = `
<nav
  className="header"
>
  <div
    className="mocked-nav-link"
  />
  <div
    className="mocked-nav-link"
  />
</nav>
`;
